src/objective-c/GRPCClient/private/NSData+GRPC.m

```diff
@@ -42,7 +42,11 @@
 static void MallocAndCopyByteBufferToCharArray(grpc_byte_buffer *buffer,
                                                size_t *length, char **array) {
   grpc_byte_buffer_reader reader;
-  grpc_byte_buffer_reader_init(&reader, buffer);
+  if (!grpc_byte_buffer_reader_init(&reader, buffer)) {
```

Comment:
Please add a comment explaining that `grpc_byte_buffer_reader_init` can fail if the data sent by the server could not be decompressed for any reason; that the server is to blame; and thus we want the RPC to fail with error code `INTERNAL`.